### ISSUE TITLE
Issue with if statement logic in ``fitting.py``

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#4858]
In the `modeling/fitting.py` file, there is the following `if` statement:

``` python
            if len(y.shape) == 2:
                rhs = y
            else:
                rhs = y
```

There was an attempt in https://github.com/astropy/astropy/pull/4692 to fix it but I don't think that's the intended meaning of the code.

@nden, since you wrote that code, could you clarify what it should be, and if possible, fix it?

**Comment by nden** [comment]
@astrofrog Originally there was some additional logic which was refactored out of the if, else leaving it like this. Since this is not changing behaviour and not really a bug I fixed it in #5788 and will close this issue when #5788 is merged.

**Comment by nden** [comment]
Fixed in #5788
